compiler/qsc/src/codegen.rs

```diff
@@ -6,18 +6,45 @@ mod tests;
 
 use qsc_codegen::qir::fir_to_qir;
 use qsc_data_structures::{language_features::LanguageFeatures, target::TargetCapabilityFlags};
-use qsc_frontend::compile::{PackageStore, SourceMap};
+use qsc_frontend::{
+    compile::{PackageStore, SourceMap},
+    error::WithSource,
+};
 use qsc_partial_eval::ProgramEntry;
 use qsc_passes::{PackageType, PassContext};
-use qsc_rca::Analyzer;
 
 use crate::compile;
 
+use miette::Diagnostic;
+use thiserror::Error;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
+pub enum Error {
```

Comment:
It would be sooo much easier for error handling up the stack for this to return a `qsc::interpret::Error` instead of this new type. The enums are almost the same, and `Interpreter::qirgen` already has to return a `qsc::interpret::Error` for these cases. Can we please unify these types?

Reply:
We already have these complicated conversion functions to turn `interpret::Error`s into VS Code diagnostics. I'd really hate to duplicate this code for another type that is _almost_ the same but not quite. 

https://github.com/microsoft/qsharp/blob/481d436395961665be75251cc32ea7b41b40caff/wasm/src/diagnostic.rs#L188